DATA_LAYER_INTEGRATION.md

```diff
@@ -269,13 +269,142 @@ The JSON rendering of a Core Component exposes a `dataLayer` property that is po
+
+    public String getMessage() {
+        return "Hello World!";
+    }
+}
+```
+
+`HelloWorld` HTL script:
+```
+<div data-sly-use.hello="com.adobe.cq.wcm.core.components.models.HelloWorld">
+    ${hello.message}
+</div>
+
+```
+
+Deploy the model and the HTL script to a running AEM instance and add this component to a page.
+Run the following code in your browser console:
+```
+adobeDataLayer.getState()
+```
+The `HelloWorld` component does not yet write to the data layer.
+
+#### Example 1: add some HelloWorld data to the data layer
+
+Add the `data-cmp-data-layer` attribute to the component HTL:
+```
+<div data-sly-use.hello="com.adobe.cq.wcm.core.components.models.HelloWorld"
+     data-cmp-data-layer="${hello.data.json}">
+    ${hello.message}
+</div>
+```
+
+Deploy the change to AEM. Refresh the page and in your browser console, get the state of the data layer:
+```
+adobeDataLayer.getState()
+```
+
+It displays something like:
+```
+helloworld-223851b702:
+    @type: "core/wcm/components/helloworld"
+    parentId: "page-611486886f"
+```
+
+#### Example 2: override the implementation of a few data properties
+
+In this example, the data of the `HelloWorld` component will inherit all its properties and implementations from the `AbstractComponentImpl`
+except for the text and the title properties which are supplied by custom implementations as follows:
+- the text property gets its value from the `getMessage()` method of the `HelloWorld` model
+- the title property gets its value from the "my title" string
+This is achieved by using the `DataLayerBuilder.extending()` API.
+
+Add following code to the `HelloWorld` model:
+```
+...
+import com.adobe.cq.wcm.core.components.models.datalayer.ComponentData;
+import com.adobe.cq.wcm.core.components.models.datalayer.builder.DataLayerBuilder;
+...
+
+    @Override
+    @NotNull
+    protected ComponentData getComponentData() {
+        return DataLayerBuilder.extending(super.getComponentData()).asComponent()
+                .withText(this::getMessage)
+                .withTitle(() -> "my title")
+                .build();
+    }
+```
+
+Build, deploy the change to AEM. Refresh the page and in your browser console, get the state of the data layer:
+```
+adobeDataLayer.getState()
+```
+
+It displays something like:
+```
+helloworld-223851b702:
+    @type: "core/wcm/components/helloworld"
+    dc:title: "my title"
+    parentId: "page-611486886f"
+    xdm:text: "Hello World!"
+```
+
+#### Example 3: add only specific data properties
+
+In this example, the data of the `HelloWorld` component will not inherit its properties from the `AbstractComponentImpl` class
+but will only use custom properties (ID, description and parent ID) and custom implementations.
+This is achieved by using the `DataLayerBuilder.forComponent()` API.
+
+Update the `getComponentData()` method of the `HelloWorld` model as follows:
+```
+    protected ComponentData getComponentData() {
```

Comment:
see previous comment

Reply:
addressed in commit https://github.com/adobe/aem-core-wcm-components/pull/1215/commits/2c7c6e594f6a35e14d19a0215406b4a871a49161